src/components/TodoModal/TodoModal.tsx

```diff
@@ -1,42 +1,70 @@
+import { getUser } from '../../api';
+import { User } from '../../types/User';
+import { useDispatch } from 'react-redux';
+import { actions } from '../../features/currentTodo';
 
-export const TodoModal: React.FC = () => {
-  return (
-    <div className="modal is-active" data-cy="modal">
-      <div className="modal-background" />
+type Props = {
+  todo: Todo;
+};
 
-      <Loader />
+export const TodoModal: React.FC<Props> = ({ todo }) => {
+  const [showLoader, setShowLoader] = useState(true);
+  const [user, setUser] = useState<User>();
+  const dispatch = useDispatch();
 
-      <div className="modal-card">
-        <header className="modal-card-head">
-          <div
-            className="modal-card-title has-text-weight-medium"
-            data-cy="modal-header"
-          >
-            Todo #3
-          </div>
+  useEffect(() => {
+    setShowLoader(true);
+    getUser(todo.userId)
+      .then(response => {
+        setUser(response);
+      })
+      .finally(() => setShowLoader(false));
+  }, [todo.userId]);
 
-          {/* eslint-disable-next-line jsx-a11y/control-has-associated-label */}
-          <button type="button" className="delete" data-cy="modal-close" />
-        </header>
+  return (
+    <div className="modal is-active" data-cy="modal">
+      <div className="modal-background" />
+      {showLoader ? (
+        <Loader />
+      ) : (
+        <div className="modal-card">
+          <header className="modal-card-head">
+            <div
+              className="modal-card-title has-text-weight-medium"
+              data-cy="modal-header"
+            >
+              Todo #{todo.id}
+            </div>
 
-        <div className="modal-card-body">
-          <p className="block" data-cy="modal-title">
-            fugiat veniam minus
-          </p>
+            {/* eslint-disable-next-line jsx-a11y/control-has-associated-label */}
+            <button
+              type="button"
+              className="delete"
+              data-cy="modal-close"
+              onClick={() => dispatch(actions.removeTodo())}
+            />
+          </header>
 
-          <p className="block" data-cy="modal-user">
-            {/* For not completed */}
-            <strong className="has-text-danger">Planned</strong>
+          <div className="modal-card-body">
+            <p className="block" data-cy="modal-title">
+              {todo.title}
+            </p>
 
-            {/* For completed */}
-            <strong className="has-text-success">Done</strong>
-            {' by '}
-            <a href="mailto:Sincere@april.biz">Leanne Graham</a>
-          </p>
+            <p className="block" data-cy="modal-user">
+              {todo.completed ? (
+                <strong className="has-text-success">Done</strong>
+              ) : (
+                <strong className="has-text-danger">Planned</strong>
+              )}
+              {' by '}
+              <a href={`mailto:${user?.email}`}>{user?.name}</a>
```

Comment:
add some spaces between elements to improve readability

Reply:
fixed